### ISSUE TITLE
Make sure to cancel all recurring contributions when archiving host

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#7701]
[This host](https://opencollective.com/mutual-aid-alnwick) was archived and continued to receive recurring payments on PayPal, the webhooks were not disconnected.

Whenever we archive a host, we must make sure that:
- All recurring contributions are canceled (incl. by calling the payment provider for externally charged contributions)
- PayPal webhooks are disconnected

**Related**
- https://open-collective.sentry.io/issues/6106184266

**Comment by Betree** [comment]
## Additional root causes (archive of independent / already-unhosted accounts)

#7701 describes archiving a host. The same leftover-PayPal-subscription failure happens when the archived account is an independent organization (or already has `HostCollectiveId = null`):

1. `archiveCollective` always calls `changeHost(null)`. If `HostCollectiveId` is already `null`, `changeHost` returns immediately and never calls `Order.stopActiveSubscriptions`. Incoming PayPal subscriptions stay `ACTIVE` in our DB and on PayPal.
2. Even when unhosting does run, `stopActiveSubscriptions` is gated on `HostCollectiveId && approvedAt`, and the SQL itself requires `c."approvedAt" IS NOT NULL`. Independent orgs receiving on their own PayPal merchant can miss both gates.
3. The follow-up `cancelActiveOrdersByCollective` only updates orders where the archived account is the **payer** (`FromCollectiveId`). Incoming contributions are untouched, and that helper does not deactivate `Subscription` or call PayPal (see #8915).
4. `deactivateMoneyManagement` (often run before archive) destroys Stripe and Wise connected accounts but leaves PayPal connected, so `/webhooks/paypal/:id` keeps receiving `PAYMENT.SALE.COMPLETED`.

The 10-minute `needsAsyncDeactivation` cron only runs if step 1 actually flagged the orders. PayPal is `isManagedExternally`, so the daily charge cron never cancels these either.

**Fix:** after `changeHost`, always `stopActiveSubscriptions` for the archived account (and children), without requiring `approvedAt`. Disconnect or ignore PayPal on money-management teardown (separate issue). Backfill leftover active PayPal subs on inactive accounts with an ops script.
